src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1991,6 +1991,42 @@ internal PointerTypeSymbol CreatePointerTypeSymbol(TypeSymbol elementType, Nulla
             return new PointerTypeSymbol(TypeWithAnnotations.Create(elementType, elementNullableAnnotation));
         }
 
+        internal FunctionPointerTypeSymbol CreateFunctionPointerTypeSymbol(
```

Comment:
>CreateFunctionPointerTypeSymbol [](start = 43, length = 31)

Is there more than one call site? Is it worth duplicating validation logic?

Reply:
Can we make it private?

---
In reply to: [428852062](https://github.com/dotnet/roslyn/pull/44436#discussion_r428852062) [](ancestors = 428852062)

Reply:
Yeah, we could. I was following the precedent from the other internal create methods. I could even go as far as removing this API and just leaving `Common`, and if we ever need to add it we can break it out again.

---
In reply to: [428894906](https://github.com/dotnet/roslyn/pull/44436#discussion_r428894906) [](ancestors = 428894906,428852062)

Reply:
> I could even go as far as removing this API and just leaving Common, and if we ever need to add it we can break it out again.

Sounds reasonable

---
In reply to: [428902442](https://github.com/dotnet/roslyn/pull/44436#discussion_r428902442) [](ancestors = 428902442,428894906,428852062)